pywr/recorders.py

```diff
@@ -248,23 +248,29 @@ def save(self):
             entry['day'] = dt.day
             entry['index'] = idx
             entry.append()
-            self._time_table.flush()
+            #self._time_table.flush()
 
         for node, ca in self._arrays.items():
             if isinstance(node, AbstractStorage):
                 ca[idx, :] = node.volume
             elif isinstance(node, AbstractNode):
                 ca[idx, :] = node.flow
             elif isinstance(node, IndexParameter):
+                a = np.empty(len(self.model.scenarios.combinations), dtype=np.int)
                 for si in self.model.scenarios.combinations:
-                    ca[idx, si.global_id] = node.index(ts, si)
+                     a[si.global_id] = node.index(ts, si)
+                ca[idx, :] = a
```

Comment:
Each call to `ca[idx, si.global_id]` creates lots of overhead in pytables. Doing it once saves this.